components/renku_data_services/project/api.spec.yaml

```diff
@@ -0,0 +1,382 @@
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/ProjectPut"
+      responses:
+        "200":
+          description: The project
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Project"
+        "404":
+          description: The project does not exist
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        default:
+          $ref: '#/components/responses/Error'
+      tags:
+        - projects
+    patch:
+      summary: Update specific fields of an existing project
+      parameters:
+        - in: path
+          name: project_id
+          required: true
+          schema:
+            type: string
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/ProjectPatch"
+      responses:
+        "200":
+          description: The project
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Project"
+        "404":
+          description: The project does not exist
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        default:
+          $ref: '#/components/responses/Error'
+      tags:
+        - projects
+    delete:
+      summary: Remove a project
+      parameters:
+        - in: path
+          name: project_id
+          required: true
+          schema:
+            type: string
+      responses:
+        "204":
+          description: The project was removed or did not exist in the first place
+        default:
+          $ref: '#/components/responses/Error'
+      tags:
+        - projects
+  /version:
+    get:
+      summary: Get the version of the service
+      responses:
+        "200":
+          description: The error
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Version"
+
+components:
+  schemas:
+    Project:
+      type: object
+      description: A Renku native project definition and metadata
+      additionalProperties: false
+      properties:
+        id:
+          $ref: "#/components/schemas/Ulid"
+        name:
+          $ref: "#/components/schemas/Name"
+        slug:
+          $ref: "#/components/schemas/Slug"
+        creation_date:
+          $ref: "#/components/schemas/CreationDate"
+        created_by:
+          $ref: "#/components/schemas/User"
+        repositories:
+          $ref: "#/components/schemas/RepositoriesList"
+        visibility:
+          $ref: "#/components/schemas/Visibility"
+        members:
+          $ref: "#/components/schemas/MembersWithRoles"
+      required:
+        - "id"
```

Comment:
I don't think this should be provided by a client. I think only "name" **really** needs to be required (slug could be generated from the name), but if prefer to get the slug from the client and just validate it, that is also fine with me.

Reply:
I thought UI is generating a slug from name since it is showing that slug to users when they edit the name. Since UI will send us a slug, I've marked it as required here.
I'll make it optional in case other clients access the API.